### ISSUE TITLE
Use of controller with CardSettingsText

### ISSUE THREAD

**Issue by anguschew** [codegrue/card_settings#93]
Hi,

Thanks for your project which is very helpful.

Everything works properly with my CardSettingsText. However, when I made use of controller (_runningTitleController) with CardSettingsText, onChanged could not be invoked.

Following is part of my codes:

```
 CardSettingsText _buildRunningLogTitle() {
    return CardSettingsText(
      controller: _runningTitleController,
     :
     :
      onChanged: (value) {
        print('******** onChanged ************* ' + value);
        setState(() {
        });
      },
```

Did I do something wrong with the codes? Thank you in advance for your help

Kind Regards,
Angus

**Comment by codegrue** [comment]
`onChanged` actually comes from the controller. The code that wires this up is `widget.controller.addListener(_handleControllerChanged)`. Does your custom controller implement an `addListener` method?

**Comment by anguschew** [comment]
Thanks for your reply. 

I have implemented an addListener method before to listen for changes. But the method defined in addListener will be invoked after CardSettingsText is constructed or the textfield is focused. This is not the expected result I want.

Therefore, if the addListener must be used, I need to write codes to check whether the text field is really changed.

**Comment by codegrue** [comment]
if you provide your own controller, this gets executes that wires up the listener, so this should work with a custom controller:

`widget.controller.addListener(_handleControllerChanged);`

_handleControllerChanged then tells the form there was a chanced value.

So this should be working out of the box. Can you post your code?

**Comment by codegrue** [comment]
Refactored the controller variables and found an issue with a custom controller not getting assigned right and not detecting a change. Fixed for release 1.9.0. Please test prior to deployment if possible.
